src/sentry/release_health/metrics.py

```diff
@@ -606,3 +604,346 @@ def extract_row_info(row: Mapping[str, Union[OrganizationId, str]]) -> ReleaseNa
+                ],
+                groupby=aggregates,
+            ),
+            referrer="release_health.metrics.get_errored_sessions_for_overview",
+        )["data"]:
+            key = row["project_id"], reverse_tag_value(org_id, row[release_column_name])
+            rv_errored_sessions[key] = row["value"]
+
+        return rv_errored_sessions
+
+    @staticmethod
+    def _get_session_by_status_for_overview(
+        where: List[Union[BooleanCondition, Condition]], org_id: int
+    ) -> Mapping[Tuple[int, str, str], int]:
+        """
+        Counts of init, abnormal and crashed sessions, purpose-built for overview
+        """
+        release_column_name = tag_key(org_id, "release")
+        session_status_column_name = tag_key(org_id, "session.status")
+
+        aggregates: List[SelectableExpression] = [
+            Column(release_column_name),
+            Column("project_id"),
+            Column(session_status_column_name),
+        ]
+
+        rv_sessions: Dict[Tuple[int, str, str], int] = {}
+
+        for row in raw_snql_query(
+            Query(
+                dataset=Dataset.Metrics.value,
+                match=Entity(EntityKey.MetricsCounters.value),
+                select=aggregates + [Column("value")],
+                where=where
+                + [
+                    Condition(Column("metric_id"), Op.EQ, metric_id(org_id, "session")),
+                    Condition(
+                        Column(session_status_column_name),
+                        Op.IN,
+                        get_tag_values_list(org_id, ["abnormal", "crashed", "init"]),
+                    ),
+                ],
+                groupby=aggregates,
+            ),
+            referrer="release_health.metrics.get_abnormal_and_crashed_sessions_for_overview",
+        )["data"]:
+            key = (
+                row["project_id"],
+                reverse_tag_value(org_id, row[release_column_name]),
+                reverse_tag_value(org_id, row[session_status_column_name]),
+            )
+            rv_sessions[key] = row["value"]
+
+        return rv_sessions
+
+    @staticmethod
+    def _get_users_and_crashed_users_for_overview(
+        where: List[Union[BooleanCondition, Condition]], org_id: int
+    ) -> Mapping[Tuple[int, str, str], int]:
+        release_column_name = tag_key(org_id, "release")
+        session_status_column_name = tag_key(org_id, "session.status")
+
+        aggregates: List[SelectableExpression] = [
+            Column(release_column_name),
+            Column("project_id"),
+            Column(session_status_column_name),
+        ]
+
+        # Count of users and crashed users
+        rv_users: Dict[Tuple[int, str, str], int] = {}
+
+        for row in raw_snql_query(
+            Query(
+                dataset=Dataset.Metrics.value,
+                match=Entity(EntityKey.MetricsSets.value),
+                select=aggregates + [Column("value")],
+                where=where
+                + [
```

Comment:
Nit: I would just append the where outside of the query and the same with cols 
Its up to you but it is kind of hard to read